example/java/TestGenomicsDB.java

```diff
@@ -21,77 +21,268 @@
+          break;
+        case 1003:
+          templateVCFHeader = g.getOptarg();
+          break;
+        case 1004:
+          lbRowIdx = Long.parseLong(g.getOptarg());
+          break;
+        case 1005:
+          ubRowIdx = Long.parseLong(g.getOptarg());
+          break;
+        case 1006:
+          chromosome = g.getOptarg();
+          break;
+        case 1007:
+          chrBegin = Integer.parseInt(g.getOptarg());
+          break;
+        case 1008:
+          chrEnd = Integer.parseInt(g.getOptarg());
+          break;
+        case 'l':
+          loaderJSONFile = g.getOptarg();
+          break;
+        case 1009:
+          countOnly = true;
+          break;
+        case 1010:
+          passAsVCF = true;
+          break;
+        default:
+          System.err.println("Unknown command line option "+g.getOptarg()+" - ignored");
+          break;
+      }
+    }
+    //Do either query or load but not both
+    assert (!doQuery || !doLoad);
+    if(!doLoad && !doQuery)
+      doQuery = true;
+    int numPositionalArgs = args.length - g.getOptind();
+    //Do query
+    if(doQuery)
+    {
+      if(passAsVCF)
+        TestGenomicsDB.<LineIterator, VCFCodec>runQuery(
+            new VCFCodec(),
+            args, g.getOptind(), numPositionalArgs,
+            loaderJSONFile,
+            workspace, array, referenceGenome, templateVCFHeader,
+            chromosome, chrBegin, chrEnd,
+            countOnly);
       else
-      if(args.length >= 9)
-        //chr,start,end specified on the command line - scan only the required positions
-        for(final VariantContext record : reader.query(args[6],
-          Integer.parseInt(args[7]), Integer.parseInt(args[8])))
-          writer.add(record);
+        TestGenomicsDB.<PositionalBufferedStream, BCF2Codec>runQuery(
+            new BCF2Codec(),
+            args, g.getOptind(), numPositionalArgs,
+            loaderJSONFile,
+            workspace, array, referenceGenome, templateVCFHeader,
+            chromosome, chrBegin, chrEnd,
+            countOnly);
     }
     else
-    if(args[0].equals("-load")) //load data
     {
+      if(numPositionalArgs == 1 && loaderJSONFile.isEmpty())
+        loaderJSONFile = args[g.getOptind()];
       //<loader.json>
-      GenomicsDBImporter loader = new GenomicsDBImporter(args[1]);
-      //Specify rank (or partition idx) of this process
-      int rank = (args.length >= 3) ? Integer.parseInt(args[2]) : 0;
-      //Specify smallest row idx from which to start loading - useful for
-      //incremental loading into existing array
-      long lbRowIdx = (args.length >= 4) ? Long.parseLong(args[3]) : 0;
-      //Specify largest row idx up to which loading should be performed - for completeness
-      long ubRowIdx = (args.length >= 5) ? Long.parseLong(args[4]) : Long.MAX_VALUE-1;
+      GenomicsDBImporter loader = new GenomicsDBImporter(loaderJSONFile);
       loader.write(rank, lbRowIdx, ubRowIdx);
     }
   }
```

Comment:
@kgururaj , this file looks like a really important one.. maybe we should move it from example to test